src/lib/graph_algorithms/nat.ml

```diff
@@ -0,0 +1,48 @@
+open Core_kernel
```

Comment:
FWIW, we also have a type level peano number implementation here https://github.com/MinaProtocol/mina/blob/develop/src/lib/gadt_lib/peano.ml. Would be nice if we could use that and add the necessary functions on top of it, but the maybe not worth dealing with an existential wrapper to erase the type parameters that other version carries.

Reply:
I think I will keep them separate. One has the phantom type and one has laziness and I think it will be a bit messy if they both have to have both.